conditions_test.go

```diff
@@ -22,3 +25,53 @@ func TestAddressPrinting(t *testing.T) {
 		So(cond.String(), ShouldNotEqual, fmt.Sprintf("%X", cond))
 	})
 }
+
+func TestAddressUnmarshalJSON(t *testing.T) {
+	cases := map[string]struct {
+		json     string
+		wantErr  error
+		wantAddr weave.Address
+	}{
+		"default decoding": {
+			json:     `"6865782d61646472"`,
+			wantAddr: weave.Address("hex-addr"),
+		},
+		"hex decoding": {
+			json:     `"hex:6865782d61646472"`,
+			wantAddr: weave.Address("hex-addr"),
+		},
+		"cond decoding": {
+			json:     `"cond:foo/bar/12:3"`,
```

Comment:
Yeah, this is a good test case, but not what I want for behavior.
In #347 I clearly specify:

Condition support: prefix cond:
* This will assume the suffix is a stringified Condition, which looks something like cond:multisig/usage/00000002 (last section of multisig is hex-encode binary)
* It will parse this data properly into a Condition (no existing code), then call Condition.Address()
* We can assert some equality, like cond.Address() == Address.UnmarshalJSON("cond:" + cond.String()) (rough pseudocode)

There is a good reason, as most desired conditions will have the last section as unprintable characters, so we should demo a hex representation.

eg. `cond:foo/bar/0123` => `weave.NewCondition("foo", "bar", []byte{0x01, 0x23}).Address()`

This also means parsing condition can produce errors, and we should validate the input is properly formatted, using logic similar to [Condition.Parse()](https://github.com/iov-one/weave/blob/master/conditions.go#L36-L46)

Reply:
Adressed in https://github.com/iov-one/weave/pull/365/commits/0275b0c008e2b7403a955b9103d445774cdd5850